Formula/autogen.rb

```diff
@@ -53,3 +53,78 @@ def install
     system bin/"autogen", "-v"
   end
 end
+
+__END__
+Index: autogen-5.18.16/agen5/guile-iface.h
+===================================================================
+--- autogen-5.18.16.orig/agen5/guile-iface.h
++++ autogen-5.18.16/agen5/guile-iface.h
+@@ -9,16 +9,13 @@
+ # error AutoGen does not work with this version of Guile
+   choke me.
+ 
+-#elif GUILE_VERSION < 203000
++#else
+ # define AG_SCM_IS_PROC(_p)           scm_is_true( scm_procedure_p(_p))
+ # define AG_SCM_LIST_P(_l)            scm_is_true( scm_list_p(_l))
+ # define AG_SCM_PAIR_P(_p)            scm_is_true( scm_pair_p(_p))
+ # define AG_SCM_TO_LONG(_v)           scm_to_long(_v)
+ # define AG_SCM_TO_ULONG(_v)          ((unsigned long)scm_to_ulong(_v))
+ 
+-#else
+-# error unknown GUILE_VERSION
+-  choke me.
+ #endif
+ 
+ #endif /* MUTATING_GUILE_IFACE_H_GUARD */
+Index: autogen-5.18.16/configure
+===================================================================
+--- autogen-5.18.16.orig/configure
++++ autogen-5.18.16/configure
+@@ -14798,7 +14798,7 @@ $as_echo "no" >&6; }
+ 		PKG_CONFIG=""
+ 	fi
+ fi
+-  _guile_versions_to_search="2.2 2.0 1.8"
++  _guile_versions_to_search="3.0 2.2 2.0 1.8"
+   if test -n "$GUILE_EFFECTIVE_VERSION"; then
+     _guile_tmp=""
+     for v in $_guile_versions_to_search; do
+Index: autogen-5.18.16/config/guile.m4
+===================================================================
+--- autogen-5.18.16.orig/config/guile.m4
++++ autogen-5.18.16/config/guile.m4
+@@ -61,7 +61,7 @@
+ #
+ AC_DEFUN([GUILE_PKG],
+  [PKG_PROG_PKG_CONFIG
+-  _guile_versions_to_search="m4_default([$1], [2.2 2.0 1.8])"
++  _guile_versions_to_search="m4_default([$1], [3.0 2.2 2.0 1.8])"
+   if test -n "$GUILE_EFFECTIVE_VERSION"; then
+     _guile_tmp=""
+     for v in $_guile_versions_to_search; do
```

Comment:
No. But we have already shipped a patch for the 2.2 version check some time ago, so the 3.0 version check disabling is almost the same: https://sourceforge.net/p/autogen/bugs/196/
Upstream told that they wanted to remove these checks completely anyway.

Also, I do not expect that much support from upstream:
https://savannah.gnu.org/support/?110504#comment1

> just note I'm retired, so I don't do this for a living anymore.

As autopen is not used that much in homebrew-core, it will probably need to be removed completely in the next years.